android/src/main/java/io/invertase/firebase/links/RNFirebaseLinks.java

```diff
@@ -0,0 +1,332 @@
+
+      DynamicLink.AndroidParameters.Builder androidParametersBuilder =
+        new DynamicLink.AndroidParameters.Builder((String) androidParameters.get("androidPackageName"));
+      androidParameters.remove("androidPackageName");
+
+      if (androidParameters.containsKey("androidFallbackLink")) {
+        androidParametersBuilder.setFallbackUrl(Uri.parse((String) androidParameters.get("androidFallbackLink")));
+        androidParameters.remove("androidFallbackLink");
+      }
+      if (androidParameters.containsKey("androidMinPackageVersionCode")) {
+        androidParametersBuilder.setMinimumVersion(Integer.parseInt((String) androidParameters.get("androidMinPackageVersionCode")));
+        androidParameters.remove("androidMinPackageVersionCode");
+      }
+
+      if (androidParameters.size() > 0) {
+        throw new IllegalArgumentException("Invalid arguments: " + androidParameters.keySet().toString());
+      }
+      parametersBuilder.setAndroidParameters(androidParametersBuilder.build());
+    }
+  }
+
+  private void setIosParameters(final Map<String, Object> dynamicLinkInfo, final DynamicLink.Builder parametersBuilder) {
+    Map<String, Object> iosParameters = (Map<String, Object>) dynamicLinkInfo.get("iosInfo");
+    if (iosParameters != null) {
+      if (!iosParameters.containsKey("iosBundleId")) {
+        throw new IllegalArgumentException("no iosBundleId was specified.");
+      }
+      DynamicLink.IosParameters.Builder iosParametersBuilder =
+        new DynamicLink.IosParameters.Builder((String) iosParameters.get("iosBundleId"));
+      iosParameters.remove("iosBundleId");
+
+      if (iosParameters.containsKey("iosAppStoreId")) {
+        iosParametersBuilder.setAppStoreId((String) iosParameters.get("iosAppStoreId"));
+        iosParameters.remove("iosAppStoreId");
+      }
+      if (iosParameters.containsKey("iosCustomScheme")) {
+        iosParametersBuilder.setCustomScheme((String) iosParameters.get("iosCustomScheme"));
+        iosParameters.remove("iosCustomScheme");
+      }
+      if (iosParameters.containsKey("iosFallbackLink")) {
+        iosParametersBuilder.setFallbackUrl(Uri.parse((String) iosParameters.get("iosFallbackLink")));
+        iosParameters.remove("iosFallbackLink");
+      }
+      if (iosParameters.containsKey("iosIpadBundleId")) {
+        iosParametersBuilder.setIpadBundleId((String) iosParameters.get("iosIpadBundleId"));
+        iosParameters.remove("iosIpadBundleId");
+      }
+      if (iosParameters.containsKey("iosIpadFallbackLink")) {
+        iosParametersBuilder.setIpadFallbackUrl(Uri.parse((String) iosParameters.get("iosIpadFallbackLink")));
+        iosParameters.remove("iosIpadFallbackLink");
+      }
+      if (iosParameters.containsKey("iosMinPackageVersionCode")) {
+        iosParametersBuilder.setMinimumVersion((String) iosParameters.get("iosMinPackageVersionCode"));
+        iosParameters.remove("iosMinPackageVersionCode");
+      }
+
+      if (iosParameters.size() > 0) {
```

Comment:
This validation code is duplicated with iOS.
See `messaging().scheduleLocalNotification()` where they validate the object in JavaScript.
I'm not sure what is more efficient (but I think we should avoid sending non relevant data over the bridge and avoid code duplication)